tests/e2e_test.go

```diff
@@ -287,3 +291,253 @@ func TestSnapshotUnvalidatedFails(t *testing.T) {
+
+	// Initialize succeeds.
+	if err := app.InitCmd(ctx, td, "", 1, targetsConfig, snapshotKey, timestampKey); err != nil {
+		t.Fatal(err)
+	}
+
+	// Sign root & targets with key 1
+	rootKey1, err := GetTestHsmSigner(ctx, td, *rootSerial1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := app.SignCmd(ctx, td, []string{"root", "targets"}, rootKey1); err != nil {
+		t.Fatal(err)
+	}
+
+	// Sign snapshot and timestamp
+	if err := app.SnapshotCmd(ctx, td); err != nil {
+		t.Fatalf("expected Snapshot command to pass, got err: %s", err)
+	}
+	snapshotSigner, err := keys.GetSigningKey(ctx, snapshotKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := app.SignCmd(ctx, td, []string{"snapshot"}, snapshotSigner); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := app.TimestampCmd(ctx, td); err != nil {
+		t.Fatalf("expected Timestamp command to pass, got err: %s", err)
+	}
+	timestampSigner, err := keys.GetSigningKey(ctx, timestampKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := app.SignCmd(ctx, td, []string{"timestamp"}, timestampSigner); err != nil {
+		t.Fatal(err)
+	}
+
+	// Successful Publishing!
+	if err := app.PublishCmd(ctx, td); err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that there are two root key signers: key 1 and key 2.
+	store := tuf.FileSystemStore(td, nil)
+	root, err := prepo.GetRootFromStore(store)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rootRole, ok := root.Roles["root"]
+	if !ok {
+		t.Fatalf("expected root role")
+	}
+	rootKey2, err := GetTestHsmSigner(ctx, td, *rootSerial2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedKeyIds := append(rootKey1.Key.IDs(), rootKey2.Key.IDs()...)
+	actualKeyIds := rootRole.KeyIDs
+	sort.Strings(expectedKeyIds)
+	sort.Strings(actualKeyIds)
+	if !cmp.Equal(expectedKeyIds, actualKeyIds) {
+		t.Fatalf("expected key IDs %s, got %s", expectedKeyIds, actualKeyIds)
+	}
+
+	// Now remove the second key and add a third key.
+	if err := os.RemoveAll(filepath.Join(td, "keys", fmt.Sprint(*rootSerial2))); err != nil {
+		t.Fatal(err)
+	}
+	rootSerial3, err := CreateTestHsmSigner(td, rootCA, rootSigner)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a new root.
+	if err := app.InitCmd(ctx, td, td, 1, targetsConfig, snapshotKey, timestampKey); err != nil {
+		t.Fatal(err)
+	}
+
+	// Check the root keys were rotated: expect key 1 and 3.
+	root, err = prepo.GetRootFromStore(store)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rootRole, ok = root.Roles["root"]
+	if !ok {
+		t.Fatalf("expected root role")
+	}
+	rootKey3, err := GetTestHsmSigner(ctx, td, *rootSerial3)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedKeyIds = append(rootKey1.Key.IDs(), rootKey3.Key.IDs()...)
+	actualKeyIds = rootRole.KeyIDs
+	sort.Strings(expectedKeyIds)
+	sort.Strings(actualKeyIds)
+	if !cmp.Equal(expectedKeyIds, actualKeyIds) {
+		t.Fatalf("expected key IDs %s, got %s", expectedKeyIds, actualKeyIds)
+	}
+
+	// Expect version 2 for root.
+	if root.Version != 2 {
+		t.Fatalf("expected root version 2, got %d", root.Version)
+	}
```

Comment:
Should we also test verification for rotations? There's two things we can test:
* With keys 1 and 2, rotate out 2 for 3, sign with 1, and test that you can still verify
* With keys 1 and 2, rotate out 1 for 3, and sign with 2, and test you can verify

Reply:
I think I can parametrize what key to rotate out for and add verification, can add!

Reply:
Feel free to do this in another PR too if it'll change non-test code.

Reply:
Adding verification -- I'll follow-up with a test using 2+ thresholds, but FWIW with threshold 1 and two keys, it doesn't matter which key I rotate out for. (I think your 2 scenarios are equivalent modulo ordering?)